app/src/main/res/layout/clock_fragment.xml

```diff
@@ -17,11 +18,29 @@
     <TextClock
             android:id="@+id/text_date_id"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
+            android:layout_height="wrap_content"
             android:format12Hour="@string/clock_format_date_12_hour"
             android:format24Hour="@string/clock_format_date_24_hour"
             android:textSize="@dimen/clock_text_date_size"
             android:textColor="@color/secondaryViewColor"
             android:gravity="center_horizontal"/>
 
+    <TextClock
```

Comment:
You have added here textClock and use it in code like TextView. Fix it.. Why you need textClock here?